operator/builtin/parser/xml/xml.go

```diff
@@ -0,0 +1,110 @@
+package xml
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+)
+
+func init() {
+	operator.Register("xml_parser", func() operator.Builder { return NewXMLParserConfig("") })
+}
+
+// NewXMLParserConfig creates a new XML parser config with default values
+func NewXMLParserConfig(operatorID string) *XMLParserConfig {
+	return &XMLParserConfig{
+		ParserConfig: helper.NewParserConfig(operatorID, "xml_parser"),
+	}
+}
+
+// XMLParserConfig is the configuration of an XML parser operator.
+type XMLParserConfig struct {
+	helper.ParserConfig `yaml:",inline"`
+}
+
+// Build will build an XML parser operator.
+func (c XMLParserConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	parserOperator, err := c.ParserConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	xmlParser := &XMLParser{
+		ParserOperator: parserOperator,
+	}
+
+	return []operator.Operator{xmlParser}, nil
+}
+
+// XMLParser is an operator that parses XML.
+type XMLParser struct {
+	helper.ParserOperator
+}
+
+// Process will parse an entry for XML.
+func (x *XMLParser) Process(ctx context.Context, entry *entry.Entry) error {
+	return x.ParserOperator.ProcessWith(ctx, entry, parse)
+}
+
+// parse will parse an xml value
+func parse(value interface{}) (interface{}, error) {
+	strValue, ok := value.(string)
+	if !ok {
+		return nil, fmt.Errorf("value is not a string")
```

Comment:
Stanza ends up chaining the error stack, so the customer will see something along the lines of: "{error: failed to parse value: value is not a string, operator_type: xml_paser, operator_id: ...}"

Reply:
Personally if I'm dealing with a parser and it says "value is not a string" it's not clear to me that it means the entire record, rather than it failed at some part in the parsing process, but it's probably an easy issue to understand once you look at the record(s) so it's not a big deal if this is the standard.

Reply:
I think I see what you mean. There exists a scenario where maybe it's one of the embedded values that is not a string, or at least a user may go down that route when they see this error message. I'll try to restructure this to be more clear.